cblas problem with self-built armadillo on GHA

#641 tests (since a looong time) building of dependent libraries. It fails (tested ubuntu 20.04) with `USE_SYSTEM_ARMADILLO=OFF`
```
 In file included from /home/runner/install/include/armadillo:54,
                 from /home/runner/install/include/gadgetron/hoArmadillo.h:10,
                 from /home/runner/install/include/gadgetron/hoNFFT.h:20,
                 from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/SIRF/src/xGadgetron/cGadgetron/include/sirf/Gadgetron/NonCartesianEncoding.h:73,
                 from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/SIRF/src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_x.h:53,
                 from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/SIRF/src/xGadgetron/cGadgetron/gadgetron_x.cpp:33:
/home/runner/install/include/armadillo_bits/compiler_setup.hpp:79:30: fatal error: /usr/include/x86_64-1-gnu/cblas.h: No such file or directory
```
https://github.com/SyneRBI/SIRF-SuperBuild/runs/4894972352?check_suite_focus=true#step:6:105

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -41,25 +41,25 @@ jobs:
           compiler: gcc
           compiler_version: 9
           BUILD_TYPE: "Release"
-          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON -DUSE_ROOT:BOOL=ON"
+          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=OFF -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON -DUSE_ROOT:BOOL=ON"
           DEVEL_BUILD: "ON"
         - os: ubuntu-latest
           compiler: gcc
           compiler_version: 8
           BUILD_TYPE: "Release"
-          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
+          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=OFF -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
           DEVEL_BUILD: "OFF"
         - os: ubuntu-18.04
           compiler: gcc
           compiler_version: 8
           BUILD_TYPE: "Release"
-          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
+          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=OFF -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
           DEVEL_BUILD: "OFF"
         - os: ubuntu-18.04
           compiler: gcc
           compiler_version: 8
           BUILD_TYPE: "Release"
-          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
+          EXTRA_BUILD_FLAGS: "-DUSE_ITK=ON -DBUILD_CIL=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=OFF -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
           DEVEL_BUILD: "ON"
 
       # let's run all of them, as opposed to aborting when one fails

--- a/docker/build_gadgetron-ubuntu.sh
+++ b/docker/build_gadgetron-ubuntu.sh
@@ -14,7 +14,7 @@ apt-get install -yq --no-install-recommends \
   h5utils                \
   jq                     \
   hdf5-tools             \
-  libatlas-base-dev      \
+  libopenblas-dev      \
   libxml2-dev            \
   libfreetype6-dev       \
   libxslt-dev            \

--- a/version_config.cmake
+++ b/version_config.cmake
@@ -151,7 +151,8 @@ set(DEFAULT_JSON_TAG v3.10.4)
 
 # CCPi CIL
 set(DEFAULT_CIL_URL https://github.com/TomographicImaging/CIL.git)
-set(DEFAULT_CIL_TAG "v21.3.1")
+# TODO update to a tagged version > v21.3.1 once available
+set(DEFAULT_CIL_TAG f5c06566215564ad78ba0f1738d0836f3ad37a60)
 set(DEFAULT_CIL-ASTRA_URL https://github.com/TomographicImaging/CIL-ASTRA.git)
 set(DEFAULT_CIL-ASTRA_TAG "v21.3.0")
 set(DEFAULT_CCPi-Regularisation-Toolkit_URL https://github.com/vais-ral/CCPi-Regularisation-Toolkit.git)

